link/src/main/java/com/stripe/android/link/LinkPaymentLauncher.kt

```diff
@@ -113,29 +116,48 @@ class LinkPaymentLauncher @AssistedInject internal constructor(
      * in during instantiation.
      *
      * @param stripeIntent the PaymentIntent or SetupIntent.
-     * @param selectedPaymentDetails the payment method previously selected by the user, if they are
-     *  returning to Link. It will be the initially selected value.
      * @param coroutineScope the coroutine scope used to collect the account status flow.
      */
     suspend fun setup(
         stripeIntent: StripeIntent,
-        selectedPaymentDetails: LinkPaymentDetails?,
         coroutineScope: CoroutineScope
     ): AccountStatus {
-        val component = setupDependencies(stripeIntent, selectedPaymentDetails)
+        val component = buildLinkPaymentLauncherComponent(stripeIntent)
         accountStatus = component.linkAccountManager.accountStatus.stateIn(coroutineScope)
         linkAccountManager = component.linkAccountManager
         linkEventsReporter = component.linkEventsReporter
+        linkComponentBuilder = component.linkComponentBuilder
+        this.stripeIntent = stripeIntent
         return accountStatus.value
     }
 
     /**
      * Launch the Link UI to process the Stripe Intent sent in [setup].
+     *
+     * @param prefilledNewCardParams The card information prefilled by the user. If non null, Link
+     *  will launch into adding a new card, with the card information pre-filled.
      */
     fun present(
-        activityResultLauncher: ActivityResultLauncher<LinkActivityContract.Args>
+        activityResultLauncher: ActivityResultLauncher<LinkActivityContract.Args>,
```

Comment:
Does this require an api review?

Reply:
No, this whole class is restricted to the library group, and only used by Payment Sheet. If we make it public eventually then it will go through one.